modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/AdminTooltip.js

```diff
@@ -92,36 +124,49 @@ const AdminTooltip = ({
 					<label htmlFor={`${namespace}pin-sku`}>
 						{Liferay.Language.get('select-sku')}
 					</label>
-					<ClayAutocomplete>
+
+					<ClayAutocomplete ref={node}>
 						<ClayAutocomplete.Input
-							onChange={(event) => setSku(event.target.value)}
-							value={sku}
-						/>
-						<ClayAutocomplete.DropDown
-							active={skus}
-							onSetActive={(event) => {
-								setSku(
-									event.target.innerText.match(/^[\S]*/g)[0]
-								);
+							onChange={(event) => {
+								setSelectedProduct(null);
+
+								setQuery(event.target.value);
+							}}
+							onFocus={() => {
+								setActive(true);
+							}}
+							onKeyUp={(event) => {
+								setActive(event.keyCode !== 27);
```

Comment:
Not a blocker, but instead of explicitly using "27" here, lets move that to a named constant for readability.

For example, this will be much easier to read for future devs rather than having to look up the  code themselves.
```js
// Add this as the top of the file
const KEY_CODE_ESCAPE = 27;
//...
setActive(event.keyCode !== KEY_CODE_ESCAPE);
```